### PR TITLE
Add aws_transfer_server table (#1909)

### PR DESCRIPTION
**Pull request by misraved**
# Integration test logs
<details>
  <summary>Logs</summary>

```
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_transfer_server []

PRETEST: tests/aws_transfer_server

TEST: tests/aws_transfer_server
Running terraform
data.aws_region.alternate: Reading...
data.aws_region.alternate: Read complete after 0s [id=us-east-1]
data.aws_region.primary: Reading...
data.aws_canonical_user_id.current_user: Reading...
data.aws_partition.current: Reading...
data.aws_caller_identity.current: Reading...
data.aws_partition.current: Read complete after 0s [id=aws]
data.aws_region.primary: Read complete after 0s [id=us-east-2]
data.aws_caller_identity.current: Read complete after 2s [id=33333333333]
data.aws_canonical_user_id.current_user: Read complete after 2s [id=d1a8fb36ce85e6f4a77eec13f53958565ce292fb0c7f8fce671bb0a9eb25dd40]
data.null_data_source.resource: Reading...
data.null_data_source.resource: Read complete after 0s [id=static]

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_transfer_server.named_test_resource will be created
  + resource "aws_transfer_server" "named_test_resource" {
      + arn                    = (known after apply)
      + domain                 = "S3"
      + endpoint               = (known after apply)
      + endpoint_type          = "PUBLIC"
      + force_destroy          = false
      + host_key_fingerprint   = (known after apply)
      + id                     = (known after apply)
      + identity_provider_type = "SERVICE_MANAGED"
      + protocols              = (known after apply)
      + security_policy_name   = "TransferSecurityPolicy-2018-11"
      + tags                   = {
          + "Name" = "turbottest50334"
        }
      + tags_all               = {
          + "Name" = "turbottest50334"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + account_id        = "33333333333"
  + aws_partition     = "aws"
  + canonical_user_id = "d1a8fb36ce85e6f4a77eec13f53958565ce292fb0c7f8fce671bb0a9eb25dd40"
  + resource_aka      = (known after apply)
  + resource_id       = (known after apply)
  + resource_name     = "turbottest50334"
aws_transfer_server.named_test_resource: Creating...
aws_transfer_server.named_test_resource: Still creating... [10s elapsed]
aws_transfer_server.named_test_resource: Still creating... [20s elapsed]
aws_transfer_server.named_test_resource: Still creating... [30s elapsed]
aws_transfer_server.named_test_resource: Still creating... [40s elapsed]
aws_transfer_server.named_test_resource: Still creating... [50s elapsed]
aws_transfer_server.named_test_resource: Still creating... [1m0s elapsed]
aws_transfer_server.named_test_resource: Still creating... [1m10s elapsed]
aws_transfer_server.named_test_resource: Still creating... [1m20s elapsed]
aws_transfer_server.named_test_resource: Still creating... [1m30s elapsed]
aws_transfer_server.named_test_resource: Still creating... [1m40s elapsed]
aws_transfer_server.named_test_resource: Still creating... [1m50s elapsed]
aws_transfer_server.named_test_resource: Still creating... [2m0s elapsed]
aws_transfer_server.named_test_resource: Still creating... [2m10s elapsed]
aws_transfer_server.named_test_resource: Still creating... [2m20s elapsed]
aws_transfer_server.named_test_resource: Still creating... [2m30s elapsed]
aws_transfer_server.named_test_resource: Still creating... [2m40s elapsed]
aws_transfer_server.named_test_resource: Still creating... [2m50s elapsed]
aws_transfer_server.named_test_resource: Creation complete after 2m54s [id=s-9dcd37a29b714dc5b]

Warning: Deprecated

  with data.null_data_source.resource,
  on variables.tf line 58, in data "null_data_source" "resource":
  58: data "null_data_source" "resource" {

The null_data_source was historically used to construct intermediate values
to re-use elsewhere in configuration, the same can now be achieved using
locals

(and one more similar warning elsewhere)

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = "33333333333"
aws_partition = "aws"
canonical_user_id = "d1a8fb36ce85e6f4a77eec13f53958565ce292fb0c7f8fce671bb0a9eb25dd40"
resource_aka = "arn:aws:transfer:us-east-2:33333333333:server/s-9dcd37a29b714dc5b"
resource_id = "s-9dcd37a29b714dc5b"
resource_name = "turbottest50334"

Running SQL query: test-get-query.sql
Warning: terminal options has been deprecated and will be removed in a future version of Steampipe.
These can now be set in a steampipe workspace.
(/Users/parthas/.steampipe/config/default.spc:34,20-42,2)
[
  {
    "akas": [
      "arn:aws:transfer:us-east-2:33333333333:server/s-9dcd37a29b714dc5b"
    ]
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
Warning: terminal options has been deprecated and will be removed in a future version of Steampipe.
These can now be set in a steampipe workspace.
(/Users/parthas/.steampipe/config/default.spc:34,20-42,2)
[
  {
    "arn": "arn:aws:transfer:us-east-2:33333333333:server/s-9dcd37a29b714dc5b",
    "server_id": "s-9dcd37a29b714dc5b"
  }
]
✔ PASSED

Running SQL query: test-resource-not-found-query.sql
Warning: terminal options has been deprecated and will be removed in a future version of Steampipe.
These can now be set in a steampipe workspace.
(/Users/parthas/.steampipe/config/default.spc:34,20-42,2)
null
✔ PASSED

POSTTEST: tests/aws_transfer_server

TEARDOWN: tests/aws_transfer_server

SUMMARY:

1/1 passed.


```
</details>

# Example query results
<details>
  <summary>Results</summary>

```
> select * from aws_transfer_server
Warning: failed to start 6 plugins using an incompatible sdk version, (required by 6 connections). To update, please run: steampipe plugin update urlscan chaos reddit zendesk ibm twilio
+--------------------------------------------------------------------+---------------------+--------+------------------------+---------------+--------------+--------+------------+-------------+------------------------------------------------------------------------------>
| arn                                                                | server_id           | domain | identity_provider_type | endpoint_type | logging_role | state  | user_count | certificate | protocol_details                                                             >
+--------------------------------------------------------------------+---------------------+--------+------------------------+---------------+--------------+--------+------------+-------------+------------------------------------------------------------------------------>
| arn:aws:transfer:us-east-2:33333333333:server/s-b7c91308852640d38 | s-b7c91308852640d38 | S3     | SERVICE_MANAGED        | PUBLIC        | <null>       | ONLINE | 0          | <null>      | {"As2Transports":null,"PassiveIp":"AUTO","SetStatOption":"DEFAULT","TlsSessio>
+--------------------------------------------------------------------+---------------------+--------+------------------------+---------------+--------------+--------+------------+-------------+------------------------------------------------------------------------------>


```
</details>

### CODE REVIEW

**Review by ParthaI — APPROVED** [review]
LGTM

File: aws/table_aws_transfer_server.go [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "endpoint_type",
+				Description: "Specifies the type of VPC endpoint that your server is connected to.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "logging_role",
+				Description: "Specifies the type of VPC endpoint that your server is connected to.",
```
Comment (madhushreeray30) [inline aws/table_aws_transfer_server.go:69]:
```suggestion
				Description: "The Amazon Resource Name (ARN) of the Identity and Access Management (IAM) role that allows a server to turn on Amazon CloudWatch logging for Amazon S3 or Amazon EFSevents. When set, you can view user activity in your CloudWatch logs.",
```

File: docs/tables/aws_transfer_server.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+```sql
+select
+  server_id,
+  domain,
+  identity_provider_type,
+  endpoint_type,
+  structured_log_destinations
+from
+  aws_transfer_server
```
Comment (madhushreeray30) [inline docs/tables/aws_transfer_server.md:68]:
```suggestion
  aws_transfer_server;
```

**Review by madhushreeray30 — COMMENTED** [review]
@misraved please have a look at the comments, thanks!

**Review by madhushreeray30 — APPROVED** [review]

**Pull request merged**